src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -312,6 +358,75 @@ private void waitForSecurityIndexToBeAtLeastYellow() {
         }
     }
 
+    // viable for testing
+    protected Future<Void> initDefaultConfiguration() {
+        return threadPool.generic().submit(() -> {
+            if (!clusterService.state().metadata().hasIndex(securityIndex)) {
+                securityIndexHandler.createIndex(
+                    ActionListener.wrap(r -> uploadSecurityConfiguration(), e -> LOGGER.error("Couldn't create index {}", securityIndex, e))
+                );
+            } else {
+                // in case index was created and cluster stat has not been changed (e.g. restart if the node or something)
+                // just upload default configuration
+                uploadSecurityConfiguration();
+            }
+            return null;
+        });
+    }
+
+    private void uploadSecurityConfiguration() {
+        securityIndexHandler.uploadSecurityConfiguration(
+            resolveConfigDir(),
+            ActionListener.wrap(
+                configuration -> clusterService.submitStateUpdateTask(
+                    "init-security-configuration",
+                    new ClusterStateUpdateTask(Priority.IMMEDIATE) {
+                        @Override
+                        public ClusterState execute(ClusterState clusterState) throws Exception {
+                            return ClusterState.builder(clusterState)
+                                .putCustom(SecurityMetadata.TYPE, new SecurityMetadata(Instant.now(), configuration))
+                                .build();
+                        }
+
+                        @Override
+                        public void onFailure(String s, Exception e) {
+                            LOGGER.error(s, e);
+                        }
+                    }
+                ),
+                e -> LOGGER.error("Couldn't upload default configuration", e)
+            )
+        );
+    }
+
+    // visible for testing
+    protected Future<Void> reloadConfiguration(final SecurityMetadata securityMetadata) {
+        if (!initalizeConfigTask.isDone() && !configReloadInProcess.get()) {
+            configReloadInProcess.getAndSet(true);
+            return threadPool.generic().submit(() -> {
+                securityIndexHandler.loadConfiguration(securityMetadata.configuration(), ActionListener.wrap(cTypeConfigs -> {
+                    configCache.putAll(cTypeConfigs);
```

Comment:
Seems like this is duplication logic from `reloadConfiguration0`, can we reduce this or make the shared usages cleaner?